src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts

```diff
@@ -1,24 +1,101 @@
+    private readonly secondsInMinute = 60;
+
+    private seconds: number;
+    private locale: string;
+    private localizedDuration: string;
+    private onLangChange?: Subscription;
+
+    constructor(private translateService: TranslateService) {}
 
-@Pipe({ name: 'artemisDurationFromSeconds' })
-export class ArtemisDurationFromSecondsPipe implements PipeTransform {
     /**
-     * Convert seconds to a human-readable duration format (mm:ss).
+     * Convert seconds to a human-readable duration format "d day(s) hh:mm:ss".
+     * The days and hours are left out if their value is zero
      * @param seconds {number}
      */
     transform(seconds: number): string {
-        const minutes = Math.floor(seconds / 60);
-        seconds = seconds - minutes * 60;
+        this.seconds = seconds;
+
+        const days = Math.floor(seconds / this.secondsInDay);
+        const hours = Math.floor((seconds % this.secondsInDay) / this.secondsInHour);
+        const minutes = Math.floor((seconds % this.secondsInHour) / this.secondsInMinute);
+        seconds = seconds % this.secondsInMinute;
+
+        let timeString = this.transformDays(days);
+
+        if (days > 0 || hours > 0) {
+            timeString += this.addLeadingZero(hours) + ':';
+        }
+
+        timeString += this.addLeadingZero(minutes) + ':';
+        timeString += this.addLeadingZero(seconds);
+
+        return timeString;
+    }
+
+    private addLeadingZero(number: number): string {
+        let numberOut = number.toString();
+        if (number < 10) {
+            numberOut = '0' + numberOut;
+        }
+        return numberOut;
+    }
+
+    private transformDays(days: number): string {
+        if (days === 0) {
+            return '';
+        }
+        // Set locale to current language.
+        this.updateLocale(this.translateService.currentLang);
+
+        // Clean up existing subscription to onLangChange and subscribe to onLangChange event, in case the language changes.
+        this.cleanUpSubscription();
+        if (!this.onLangChange) {
+            this.onLangChange = this.translateService.onLangChange.subscribe((event: LangChangeEvent) => this.updateLocale(event.lang));
+        }
+
+        return days + this.getDayString(days);
+    }
+
+    private getDayString(days: number): string {
+        if (this.locale === 'de') {
+            return days > 1 ? ' Tage ' : ' Tag ';
+        } else {
+            return days > 1 ? ' days ' : ' day ';
+        }
+    }
 
-        let minutesOut = minutes.toString();
-        if (minutes < 10) {
-            minutesOut = '0' + minutes.toString();
+    private updateLocale(lang: string): void {
+        if (this.locale === undefined && lang === undefined) {
+            this.locale = 'en';
         }
-        let secondsOut = seconds.toString();
-        if (seconds < 10) {
-            secondsOut = '0' + seconds.toString();
+        if (lang !== this.locale) {
+            this.locale = lang;
```

Comment:
This part here looks rather odd. If `lang` and `locale` are `undefined`, we will first set `locale` to `en`, then compare `en` to `undefined`, which are different, and thus set the `locale` back to `undefined`?